### ISSUE TITLE
🚀 Feature: Add a no-file-dependencies rule

### ISSUE THREAD

**Issue by AlexMan123456** [JoshuaKGoldberg/eslint-plugin-package-json#1420]
### Feature Request Checklist

- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

As of now, it is possible to pin a package to a local file directory, which is often useful for local development but most likely unintentional if left in in production code. For example, I may have something like

```
"dependencies": {
    "@alextheman/components": "file:../components"
}
```

in one of my repositories to try out my local components package in a separate project I'm working in (this project and my components package are different repositories). However, when committing changes, we should instead be using the live version of the package and therefore pin it to an actual version number. For example:

```
"dependencies": {
    "@alextheman/components": "5.9.0"
}
```

It would be nice if I could enforce this with an ESLint rule of some sort, such that if a dependency is pinned to something including `file:`, the rule is triggered. Optionally, we could include a `target` option that takes an array of strings, where each item can either be "dependencies", "devDependencies", and/or "peerDependencies", where only the dependency categories listed will be targeted by the rule (by default, if no options provided it will target all dependency groups).

**Comment by michaelfaith** [comment]
Nice idea.  ~And I could see this going in the `recommended-publishable` config.~  I'd recommend that we keep it simple and just enforce for `dependencies`.

**Comment by AlexMan123456** [comment]
> Nice idea.  And I could see this going in the `recommended-publishable` config.  I'd recommend that we keep it simple and just enforce for `dependencies`. 

@michaelfaith Yeah, sounds good to me! I suppose we can just start with checking dependencies for now, and then maybe extend the rule to allow checking other dependency categories in the future. I look forward to seeing this implemented!

**Comment by AlexMan123456** [comment]
> > Also, what are your thoughts on wildcard dependencies? 
> 
> It's not something I would personally do, but I've definitely seen packages use * for `peerDependencies` before, in cases that I thought were fairly reasonable.  I imagine there are probably some that do that for `dependencies`.  Seems less legitimate in the case of `dependencies`, but I think there's a difference between "hey that was probably a mistake and we should flag it" (in the case of local file dependencies) vs people just making "interesting" choices 😅.  The former I would feel comfortable including in our recommended publishable config.  The latter I probably wouldn't.  
> 
> That's not to say there isn't room for a different rule for that, but I wouldn't want to conflate that with this.
> 

Yeah, that's fair. In that case, let's just focus on file dependencies and maybe also link for this.
